### DISCUSSION TITLE
How do I set up a GitHub Actions workflow to automatically deploy a Docker application

### DISCUSSION THREAD

**Discussion by meobietcode** [community/community#139087] [Actions] [answered]
### Select Topic Area

Question

### Body

How do I set up a GitHub Actions workflow to automatically deploy a Docker application to a Kubernetes cluster, but only when the code changes are pushed to the main branch and have been approved by at least two people in a pull request?

**Comment by ngchthien** [comment, marked as answer]
You can use a combination of GitHub Actions features to achieve this:
name: Deploy the app

on:
pull_request:
branches:
- main
types: [closed]

jobs:
check-approvals:
runs-on: ubuntu-latest
if: github.event.pull_request.merged == true && github.event.pull_request.state == 'closed'
steps:
- name: Check the number of approvers
uses: pullreminders/github-action-required-approvals@v1.0
with:
required_approvals: 2
env:
GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

deploy:
needs: check-approvals
runs-on: ubuntu-latest
if: ${{ needs.checkout.outputs.approved == 'true' }}

steps:
- name: Checkout code
uses: actions/checkout@v3

# Steps to build and push your Docker image
# ...

- name: Deploy to Kubernetes
uses: some-kubernetes-action@v1
with:
# Configure your Kubernetes deployment
# ...
env:
KUBECONFIG: ${{ secrets.KUBECONFIG }}
Use code with caution.
Yaml
Explanation:
Trigger event: Workflow triggered when a pull request is merged into the main branch.
Check-Approval job:
Check if the pull request has been merged and closed.
Use the pullreminders/github-action-required-approvals action to check the minimum number of approvers (2 in this case).
Save the check result to the approved output.
Deployment job:
needs: approved-check: Ensures this job only runs after the approved-check job completes.
if: ${{ needs.approved-check.outputs.approved == 'true' }}: Only run this job if the pull request has been approved by enough people.
Perform the steps to build the Docker image, push to the registry, and deploy to Kubernetes.
Note:
Replace some-kubernetes-action@v1 with the action that matches your Kubernetes deployment needs.
Configure KUBECONFIG and other sensitive information using GitHub Secrets.
